ext/standard/basic_functions.stub.php

```diff
@@ -2056,7 +2098,10 @@ function crc32(string $string): int {}
 
 /* crypt.c */
 
-/** @refcount 1 */
+/**
+ * @compile-time-eval
```

Comment:
`crypt()` is designed to be slow. I don't find it appropriate to try to evaluate that at compile time. The same is likely true for all the cryptographic hash functions.

Reply:
I will remove `crypt()`. `hash()` should be fast and common algos like `sha256` should be definitely supported to be CTEable like `md5()`. Does `hash()` support any really slow algos? Also note, if the input is not a const, nothing is evaluted as compile time...

Reply:
I don't see the point, when you have a constant string... For arbitrary inputs, sure.

Reply:
> I don't see the point to not eval it, when you have a constant string...

It makes compilation (much) slower for a function call that might not be necessary for the vast majority of requests, thus possibly breaking response time SLAs for unrelated requests.

Reply:
I think any function doing significant computations shouldn't be added to CTE, like hashing or expensive math, e.g. `hash`, `expm1`. Constant parameters for these are mostly occurring in test files, where opcache is more likely to be disabled and there is no high traffic to benefit from CTE. Also given there are already segfaults for hash in this PR, this could become security issue after 8.3 release.

Reply:
hash functions are fast, the same for common math function, both should be perfectly usable in constant expressions

Reply:
> hash functions are fast

Cryptographic hash functions definitely are on the slower end of the spectrum. Hashing a 10kB string with SHA-256 takes roughly 0.1ms on my machine. Not much by itself, but this can quickly add up.

The comment for ZEND_SUPPORTS_COMPILE_TIME_EVAL_FE specifically states:

https://github.com/php/php-src/blob/648e896d0eb373187582c77715c5b5e7a57005e3/Zend/zend_API.h#L107-L108

I consider 10kB inputs to be something that may be seen in practice (the limit for CTE `str_repeat` is 64kB output) and I also consider 0.1ms to optimize a single function call to be unreasonable. Especially since delays like this are hard to debug, because they happen before the script actually executes.

Reply:
Thank you for providing a real data, even I do not expect long data to be present as const and hashed, I agree many hashes in a class would slow the compile time too much... Reverting.